flocker/node/functional/test_docker.py

```diff
@@ -356,6 +398,7 @@ def image_built(image_name):
                 unit_name=unit_name,
                 image_name=image_name,
                 environment=Environment(variables=expected_variables),
+                force=True,
```

Comment:
Also, what about other tests that create and then remove containers (ie, all the other tests using `start_application`)?

Couldn't they all also fail with this error?  The error doesn't seem particularly container-with-environment-variables specific at first glance.  If any `start_container`-using test could encounter this problem, it would probably make more sense to just have force removal be the behavior `start_container` implements and not bother with the parameter and updating all the callers.